Engine/Helper.cs

```diff
@@ -604,6 +605,184 @@ public bool HasSpecialVars(string varName)
             return false;
         }
 
+        /// <summary>
+        /// Returns a dictionary of rule suppression from the ast.
+        /// Key of the dictionary is rule name.
+        /// Value is a list of tuple of integers that represents the interval to apply the rule
+        /// </summary>
+        /// <param name="ast"></param>
+        /// <returns></returns>
+        public Dictionary<string, List<RuleSuppression>> GetRuleSuppression(Ast ast)
+        {
+            List<RuleSuppression> ruleSuppressionList = new List<RuleSuppression>();
+
+            ScriptBlockAst sbAst = ast as ScriptBlockAst;
+
+            // Get rule suppression from the ast itself if it is scriptblockast
+            if (sbAst != null && sbAst.ParamBlock != null && sbAst.ParamBlock.Attributes != null)
+            {
+                ruleSuppressionList.AddRange(RuleSuppression.GetSuppressions(sbAst.ParamBlock.Attributes, sbAst.Extent.StartOffset, sbAst.Extent.EndOffset));
+            }
+            
+            // Get rule suppression from functions
+            IEnumerable<FunctionDefinitionAst> funcAsts = ast.FindAll(item => item is FunctionDefinitionAst, true).Cast<FunctionDefinitionAst>();
+
+            foreach (var funcAst in funcAsts)
+            {
+                ruleSuppressionList.AddRange(GetSuppressionsFunction(funcAst));
+            }
+
+            // Get rule suppression from classes
+            IEnumerable<TypeDefinitionAst> typeAsts = ast.FindAll(item => item is TypeDefinitionAst, true).Cast<TypeDefinitionAst>();
+
+            foreach (var typeAst in typeAsts)
+            {
+                ruleSuppressionList.AddRange(GetSuppressionsClass(typeAst));
+            }
+
+            Dictionary<string, List<RuleSuppression>> results = new Dictionary<string, List<RuleSuppression>>(StringComparer.OrdinalIgnoreCase);
+            ruleSuppressionList.Sort((item, item2) => item.StartOffset.CompareTo(item2.StartOffset));
+
+            foreach (RuleSuppression ruleSuppression in ruleSuppressionList)
+            {
+                if (!results.ContainsKey(ruleSuppression.RuleName))
+                {
+                    List<RuleSuppression> ruleSuppressions = new List<RuleSuppression>();
+                    results.Add(ruleSuppression.RuleName, ruleSuppressions);
+                }
+                
+                results[ruleSuppression.RuleName].Add(ruleSuppression);
+            }
+
+            return results;
+        }
+        
+        /// <summary>
+        /// Returns a list of rule suppressions from the function
+        /// </summary>
+        /// <param name="funcAst"></param>
+        /// <returns></returns>
+        internal List<RuleSuppression> GetSuppressionsFunction(FunctionDefinitionAst funcAst)
+        {
+            List<RuleSuppression> result = new List<RuleSuppression>();
+
+            if (funcAst != null && funcAst.Body != null
```

Comment:
I'm sure it's not going to be null. But I have to add this check for the OACR ...